crates/sui-framework/packages/bridge/sources/treasury.move

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module bridge::treasury {
+    use std::type_name;
+
+    use sui::coin::{Self, Coin};
+    use sui::object_bag::{Self, ObjectBag};
+    use sui::sui::SUI;
+    use sui::tx_context::{Self, TxContext};
+
+    use bridge::btc;
+    use bridge::btc::BTC;
+    use bridge::eth;
+    use bridge::eth::ETH;
+    use bridge::usdc;
+    use bridge::usdc::USDC;
+    use bridge::usdt;
+    use bridge::usdt::USDT;
+
+    const EUnsupportedTokenType: u64 = 0;
+    const ENotSystemAddress: u64 = 1;
+
+    struct BridgeTreasury has store {
+        treasuries: ObjectBag
+    }
+
+    public fun create(ctx: &mut TxContext): BridgeTreasury {
+        assert!(tx_context::sender(ctx) == @0x0, ENotSystemAddress);
+        BridgeTreasury {
+            treasuries: object_bag::new(ctx)
+        }
+    }
+
+    public fun burn<T>(self: &mut BridgeTreasury, token: Coin<T>, ctx: &mut TxContext) {
+        create_treasury_if_not_exist<T>(self, ctx);
+        let treasury = object_bag::borrow_mut(&mut self.treasuries, type_name::get<T>());
+        coin::burn(treasury, token);
+    }
+
+    public fun mint<T>(self: &mut BridgeTreasury, amount: u64, ctx: &mut TxContext): Coin<T> {
+        create_treasury_if_not_exist<T>(self, ctx);
+        let treasury = object_bag::borrow_mut(&mut self.treasuries, type_name::get<T>());
+        coin::mint(treasury, amount, ctx)
+    }
+
+    fun create_treasury_if_not_exist<T>(self: &mut BridgeTreasury, ctx: &mut TxContext) {
```

Comment:
correct! we "only" need to upgrade framework code (and fix all the OTW complains) when we need to add new currency, don't need EndOfEpoch TX to create the TreasuryCap, it will be created lazily.